internal/p2p/applicator.go

```diff
@@ -310,14 +315,22 @@ func (b *Applicator) handleApplyBlock(request *blockApplicationRequest) {
 		err = p2perrors.ErrBlockIrreversibility
 	} else {
 		_, err = b.rpc.ApplyBlock(request.ctx, request.block)
+		if err == nil {
+			b.cache.CheckBlock(request.block)
```

Comment:
This will already be handled by handling the block broadcast. Do we want to do the same work twice?